services/filemanager/filemanager.go

```diff
@@ -87,8 +87,27 @@ func (factory *FileManagerFactoryT) New(settings *SettingsT) (FileManager, error
 	return nil, fmt.Errorf("%w: %s", rterror.InvalidServiceProvider, settings.Provider)
 }
 
-// GetProviderConfigForBackupsFromEnv returns the provider config
-func GetProviderConfigForBackupsFromEnv(ctx context.Context) map[string]interface{} {
+// GetBackupStorageConfig returns the backup storage config based on the user settings
+func GetBackupStorageConfig(ctx context.Context) (string, map[string]interface{}) {
+	backendconfig.DefaultBackendConfig.WaitForConfig(ctx)
+	// TODO: Adapt for multi-tenancy
+	workspaceID := backendconfig.DefaultBackendConfig.GetWorkspaceIDForWriteKey("")
+	useRudderStorage := backendconfig.DefaultBackendConfig.GetDataRetentionSettingsForWorkspaceID(workspaceID).UseRudderServerStorage
+
+	if useRudderStorage {
+		return getProviderAndConfigForBackupsFromEnv()
+	} else {
+		return getProviderAndConfigForBackupsFromWorkspaceSettings(backendconfig.DefaultBackendConfig.GetDataRetentionSettingsForWorkspaceID(workspaceID).StorageBucket)
```

Comment:
Why getting (thus Rlocking) twice here?

```suggestion
	retentionSettings := backendconfig.DefaultBackendConfig.GetDataRetentionSettingsForWorkspaceID(workspaceID)
	useRudderStorage := retentionSettings.UseRudderServerStorage

	if useRudderStorage {
		return getProviderAndConfigForBackupsFromEnv()
	} else {
		return getProviderAndConfigForBackupsFromWorkspaceSettings(retentionSettings.StorageBucket)
```

Reply:
done